### PR TITLE
[JENKINS-68381] allow to set id and style for l:icon with symbols

### PR DESCRIPTION
**Pull request by mawinter69**
See [JENKINS-68381](https://issues.jenkins.io/browse/JENKINS-68381).

when using <l:icon src="symbol-symbol" id="id"/> the id was not
transferred to the svg.
Similar for style.
This change fixes this and also forwards these attributes when
iconMetadata.svgSprite evaluates to true

See [JENKINS-XXXXX](https://issues.jenkins.io/browse/JENKINS-XXXXX).

### Proposed changelog entries

* Allow setting an "id" and style attribute for `l:icon` in combination with symbols.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/org/jenkins/ui/icon/IconSet.java
Comment (mawinter69) [inline core/src/main/java/org/jenkins/ui/icon/IconSet.java:126]: This is inconsistent to line 93 and the test.
When a symbol loaded from the resource already contains a tooltip but no tooltip was set in the method call that would not be properly removed here.

File: core/src/main/java/org/jenkins/ui/icon/IconSet.java
Comment (mawinter69) [inline core/src/main/java/org/jenkins/ui/icon/IconSet.java:125]: Why keep an empty class attibute?
The result is when classes are passed
that we have 
`<svg class="class1 class" class="" ...>`

File: core/src/main/java/org/jenkins/ui/icon/IconSet.java
Comment (timja) [inline core/src/main/java/org/jenkins/ui/icon/IconSet.java:84]: do you need style? inline style isn't recommended (that's the main thing holding me back from approving this).

Also what do you want the ID for?

Comment (mawinter69) [reply core/src/main/java/org/jenkins/ui/icon/IconSet.java:84]: For #6498 I want to use an a clickable chevron up/down to display additional labels. To be able to change visibility of the icon I need it's id.

The style was just for completeness as the icon taglib explicitly allows to set a style, but I do not need it for my purposes

Comment (timja) [reply core/src/main/java/org/jenkins/ui/icon/IconSet.java:84]: could you remove the style please?

do you need id or could you use class? most of Jenkins works by using class selector's so that JavaScript is applied to all matching elements, but we can add ID if you need it

Comment (mawinter69) [reply core/src/main/java/org/jenkins/ui/icon/IconSet.java:84]: in my use case I need to identify the icon for a certain node on the /computer page. To use the class I would need to create a class containing the computer name, but that can probably lead to problems. Using an ID is I think the cleaner way

File: core/src/main/resources/lib/layout/icon.jelly
Comment (timja) [inline core/src/main/resources/lib/layout/icon.jelly:91]: unrelated change?

Comment (mawinter69) [reply core/src/main/resources/lib/layout/icon.jelly:91]: sort of. but all other places pass on the tooltip so why not there as well?

Comment (timja) [reply core/src/main/resources/lib/layout/icon.jelly:91]: not sure, i think it should be split out to another PR, at least the build status one works but it might have special handling.

Have you got a use-case for this?

Comment (mawinter69) [reply core/src/main/resources/lib/layout/icon.jelly]: removed passing the tooltip as I don't need it

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
